### DISCUSSION TITLE
Change current animation time

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#4831] [Q&A] [answered]
Hello,

A piece of code that was working fine to change the animation time:
```Typescript
viewport.displayStyle.settings.timePoint = value
```
Now no longer works, meaning the model doesn't change accordingly, also tried this and doesn't work too:
```Typescript
viewport.analysisFraction = fraction
```

What is the correct way to change the current timeline time, given that the default animation player does the job correctly, and it appears to be using analysisFraction 
https://github.com/iTwin/itwinjs-core/blob/e0a47b42ff521849957a14df863a86252f791487/ui/appui-react/src/appui-react/timeline/AnalysisAnimationProvider.ts#L53

**Comment by pmconne** [comment, marked as answer]
Hello.

The animation player uses a different provider based on the view. `AnalysisAnimationProvider` is used for scientific visualization, which is not relevant here. For schedule scripts, it uses [ScheduleAnimationProvider](https://github.com/iTwin/itwinjs-core/blob/e0a47b42ff521849957a14df863a86252f791487/ui/appui-react/src/appui-react/timeline/ScheduleAnimationProvider.ts#L60), which sets `Viewport.timePoint`.

You say your first line of code "was working" and now "no longer works".
Can you please specify in which version of iTwin.js it worked, and in which version it no longer works?

Thanks.

**Reply by salah-walid** [reply]
We were using version 3.1.3, updated it to 3.5.1 just to check, istill doesn't work

**Reply by pmconne** [reply]
It worked in 3.1.3, then you upgraded to 3.5.1 and it stopped working?
We have tests for this stuff - they haven't caught any regressions.
Did you change anything else when upgrading?
Verified in debugger that your script has a schedule, the value you're passing to `timePoint` is within range, etc?
After changing `timePoint` you should see breakpoint hit [here](https://github.com/iTwin/itwinjs-core/blob/782ff92243a6bc1e8244a270faf848208a09ae5d/core/frontend/src/Viewport.ts#L2338).

**Reply by salah-walid** [reply]
I figured it out, apparently using a viewport generated from IModelApp.viewManager.onViewOpen.addOnce you can't change the timePoint,  switching to using IModelApp.viewManager.selectedView solved the issue, is that normal ?

**Reply by pmconne** [reply]
Doesn't sound normal. I don't use the Viewer component, or React for that matter. I recall for a long time it had issues where it'd open a first viewport, then throw that away and open a second one, causing anybody listening for `onViewOpen.addOnce` to interact with the "wrong" viewport. Are you using the Viewer component?

Might be worth setting breakpoints in `ViewManager.addViewport` and `ViewManager.dropViewport` to see if more than one viewport is being registered, and if so, from where (call stack).

However depending on your app/tool it may not make sense to assume only one viewport ever gets created. For example, when switching between front-stages I think you can expect a new Viewport.

@kckst8 @aruniverse @NancyMcCallB ?

**Reply by salah-walid** [reply]
> However depending on your app/tool it may not make sense to assume only one viewport ever gets created. For example, when switching between front-stages I think you can expect a new Viewport.

We are using a simple Frontstage with one simple viewport, also yes we are using the react Viewer
